File: python/easy/move_zeros.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        if len(nums) <= 1:
            return
        index = 0
        travel = 0
        while(travel < len(nums)):
            if nums[travel] != 0 :
                tmp = nums[index]
                nums[index] = nums[travel]
                nums[travel] = tmp
                index = index+1
            travel = travel+1
        
        

File: cpp/easy/move_zeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        if(nums.size() <= 1)
            return;
        int curr = 0;
        int tmp = 0;
        for(int i = 0; i < nums.size(); ++i)
        {
            if(nums[i] != 0)
            {
                tmp = nums[i];
                nums[i] = nums[curr];
                nums[curr] = tmp;
                ++curr;
            }
        }
    }
};

Conclusion:

easy issue: no extra comments.

time complexity: O(n)
space complexity: O(1)
